### DISCUSSION TITLE
Wikilink conversion

### DISCUSSION THREAD

**Discussion by owenvachell** [adityatelange/hugo-PaperMod#435] [Ideas]
Love the theme. Would like for wikilinks ([[markdown filename]]) to be automatically converted. [this reddit post](https://www.reddit.com/r/gohugo/comments/gqak1j/transform_wikilinks_in_markdown/?utm_source=share&utm_medium=ios_app&utm_name=iossmf) will help demonstrate what I mean.

I spotted some regex in the above thread but have no idea what to do with it. If you're able to point me in the right direction I'd greatly appreciate it.

This is the regex:
```
{{ .Content | replaceRE "\\[\\[([^/]+)\\]\\]" "<a href=\"${1}.html\">$1</a>" |safeHTML }}
```

I tried it in the anchored_headings.html file. It converted to a url but only added the filename in the link to the end of the url for the current page.

**Comment by adityatelange** [comment]
Hi @owenvachell, I think it should work with `.RawContent` 

```
{{ .RawContent | replaceRE "\\[\\[([^/]+)\\]\\]" "<a href=\"${1}.html\">$1</a>" | safeHTML }}
```
`.Content` is already rendered HTML and not markdown text, using `.RawContent` gives you raw markdown text.

**Reply by owenvachell** [reply]
Where would I put this in the template? I couldn't find anything that looked like URL formatting. Or have you done it already? Please let me know if you have so that I can pull an update.

**Reply by adityatelange** [reply]
I have not added it. IMO it would be better if it could be added in HUGO directly. An issue is already open [here](https://github.com/gohugoio/hugo/issues/3606)

**Reply by meleu** [reply]
@adityatelange could you point where in the theme files that snippet should be added?

Even if it's cleaner/better to have support to [[wikilinks]] directly on HUGO, while it doesn't happen, having a temporary workaround would be awesome for people who use software like [Obsidian](https://obsidian.md/) to edit their markdown files.

**Note**: such temporary workaround is to be done on our side, not to be commited in your code.

**Reply by adityatelange** [reply]
Here:
https://github.com/adityatelange/hugo-PaperMod/blob/80928d9f2327809b7ecccf36930e4b726a7d10b2/layouts/_default/single.html#L34

https://github.com/adityatelange/hugo-PaperMod/wiki/FAQs#override-theme-template
